src/libraries/System.Private.Xml/src/System/Xml/XmlResolver.ThrowingResolver.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Threading.Tasks;
+
+namespace System.Xml
+{
+    public abstract partial class XmlResolver
+    {
+        /// <summary>
+        /// Gets an <see cref="XmlResolver"/> which forbids entity resolution.
+        /// </summary>
+        /// <remarks>
+        /// Calling <see cref="GetEntity"/> or <see cref="GetEntityAsync"/> on the
+        /// <see cref="XmlResolver"/> instance returned by this property is forbidden
+        /// and will result in <see cref="XmlException"/> being thrown.
+        ///
+        /// Use <see cref="ThrowingResolver"/> when external entity resolution must be
+        /// prohibited, even when DTD processing is otherwise enabled.
+        /// </remarks>
+        public static XmlResolver ThrowingResolver => XmlThrowingResolver.Singleton;
+
+        // An XmlResolver that forbids all external entity resolution.
+        private sealed class XmlThrowingResolver : XmlResolver
+        {
+            internal static XmlThrowingResolver Singleton = new();
```

Comment:
As a static field, it'd be `s_singleton`.  Since there's only one access of it, may as well be a field.

```suggestion
            internal static readonly XmlThrowingResolver s_singleton = new();
```